sdk-java/src/main/java/io/littlehorse/sdk/common/auth/OAuthCredentialsProvider.java

```diff
@@ -23,10 +23,10 @@ public void applyRequestMetadata(RequestInfo requestInfo, Executor executor, Met
         executor.execute(() -> {
             try {
                 if (currentToken == null || currentToken.isExpired()) {
-                    log.debug("Token expired, requesting a new one");
                     currentToken = oauthClient.getAccessToken();
+                    log.debug("Token was expired. New token: {}", currentToken);
```

Comment:
@sauljabin is it safe to log the token? I mean, the token is temporary and the access can be revoked but I'm not sure if it would be consider a security breach.

Reply:
Definitely don't `debug` log it. At most a `trace`. And yes, I would rather not log it at all.

Reply:
good catch
1 All this logs should be removed before we finish (not only the token itself), I think they do not provide to much value
2 I was running into a bug, so we need unit test on this, in python i'm trying to cover as much as i can
3 This is the toString:
```
public String toString() {
        return String.format("IntrospectResponse [expired=%s, expiration date=%s]", isExpired(), expiration);
        return String.format(
                "TokenStatus [client id=%s, expired=%s, expiration date=%s]", clientId, isExpired(), expiration);
    }
```
We are not printing the token itself, just the status, i just needed to see that info not the token